test/generate-plugin.test.js

```diff
@@ -100,16 +100,35 @@ function define (t) {
   })
 
   test('should finish succesfully', async (t) => {
-    t.plan(17 + Object.keys(expected).length)
+    t.plan(18 + Object.keys(expected).length)
     try {
       await generate(workdir, pluginTemplate)
       await verifyPkg(t)
       await verifyCopy(t, expected)
+      await verifyFastifyPluginVersion(t)
     } catch (err) {
       t.error(err)
     }
   })
 
+  function verifyFastifyPluginVersion (t) {
+    return new Promise((resolve, reject) => {
+      const pkgFile = path.join(workdir, 'package.json')
+
+      readFile(pkgFile, function (err, data) {
+        err && reject(err)
+        const pkg = JSON.parse(data)
+        const indexFilePath = path.join(workdir, 'index.js')
+        const indexFile = readFileSync(indexFilePath)
+        const version = pkg.dependencies['fastify-plugin']
+        const majorToMatch = version.split('.')[0]
+        const foundCorrectFastifyVersion = indexFile.toString().indexOf(`fastify: '${majorToMatch}.x'`) >= 0
+        t.equal(foundCorrectFastifyVersion, true)
```

Comment:
Instead of this `indexOf` check, we may:

- load the `indexFile` as a module (`const plugin = require(indexFile)`)
- since it is a function we can read the options with `plugin[Symbol.for('plugin-meta')]`
- then we can use the [`semver` dep](https://www.npmjs.com/package/semver) (hopefully it is already installed, or we can add it as devDep) to check if the versions matches